### ISSUE TITLE
Not working on Linux

### ISSUE THREAD

**Issue by kypi** [UnfamiliarLegacy/G-Earth#2]
Get RC4 table error when attempting to run on Linux

**Comment by detolly** [comment]
I'm working on a separate implementation if you're interested:

I've separated out the privileged operations into separate processes because it simply doesnt work in wayland. Doesn't work yet though

https://github.com/detolly/G-PrivilegedOperations
https://github.com/detolly/G-Earth

**Comment by detolly** [comment]
Works now

https://github.com/detolly/G-Earth/releases/tag/1.5.4-beta-8-linux

**Comment by UnfamiliarLegacy** [comment]
Copying what I wrote on discord

---

I looked into supporting linux but it's a pain.

**1. Automatically installing CA certificates differs per linux distro.**
i.e. ubuntu: <https://askubuntu.com/a/94861>, arch: <https://wiki.archlinux.org/title/Transport_Layer_Security#Add_a_certificate_to_a_trust_store>, fedora: <https://docs.fedoraproject.org/en-US/quick-docs/using-shared-system-certificates/#proc_adding-new-certificates>

**2. Registering a system proxy is a pain programmatically**
For example ubuntu <https://phoenixnap.com/kb/ubuntu-proxy-settings>
tldr; either use the networking gui interface, or modify the `~/.profile` / `~/.bashrc` / `~/.zshrc`, so this also depends on the shell used.

I don't want to put code into G-Earth that messes with someone their root certificate files directly or their bash profiles. Plus this would require **root (sudo)**, so, no.

--

Another option is to let linux users set both up themselves but this will require some G-Earth changes
1. Remove port randomization for the http proxy (or make it configurable)
2. Keep the http proxy running when connected to the hotel, because g-earth cant unregister the http proxy for you. This can have unknown consequences on the http traffic going through your machine because it will keep using G-Earth to proxy it

**Comment by Fadhly-leaf** [comment]
This happened to MacOS too. i kept on getting RC4 table error
